rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -2563,9 +2582,9 @@ main_window::drop_type main_window::IsValidFile(const QMimeData& md, QStringList
 
 			drop_type = drop_type::drop_pkg;
 		}
-		else if (info.suffix() == "rap")
+		else if (info.suffix().toLower() == "rap")
 		{
-			if (drop_type != drop_type::drop_rap && drop_type != drop_type::drop_error)
+			if (info.size() < 0x10 || (drop_type != drop_type::drop_rap && drop_type != drop_type::drop_error))
```

Comment:
why check for < 0x10 here when you check for != 0x10 in other places?

Reply:
fs::file::read is allowed to return values less than full file size if we did not request to read all its contents.

Reply:
It's weird but some RAP files are 16 bytes in size and others exactly 23, no idea why it's like that and if there are more versions for its size.